File: app/src/main/java/com/example/sasham/testproject/dependencies/ThemesModule.java
package com.example.sasham.testproject.dependencies;

import com.example.sasham.testproject.messages.MessagesFragment;
import com.example.sasham.testproject.themes.ThemesListingFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public interface ThemesModule {

    @ThemesListingFragmentScope
    @ContributesAndroidInjector(modules = ThemesListingFragmentModule.class)
    ThemesListingFragment themesListingFragmentInjector();

}

File: app/src/main/java/com/example/sasham/testproject/dependencies/ThemesModule.kt
package com.example.sasham.testproject.dependencies

import com.example.sasham.testproject.themes.ThemesFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
interface ThemesModule {

    @PerFragment
    @ContributesAndroidInjector(modules = arrayOf(ThemesListingFragmentModule::class))
    fun themesFragment(): ThemesFragment
}

@Module
interface ThemesListingFragmentModule {

}
